api/include/opentelemetry/metrics/meter.h

```diff
@@ -0,0 +1,204 @@
+   * @throws IllegalArgumentException if a different metric by the same name exists in this meter.
+   * @throws IllegalArgumentException if the {@code name} does not match spec requirements.
+   */
+  virtual opentelemetry::nostd::shared_ptr<AsynchronousInstrument<T>> NewUpDownSumObserver(
+      nostd::string_view name,
+      nostd::string_view description,
+      nostd::string_view unit,
+      const bool enabled,
+      void (*callback)(ObserverResult<T>)) = 0;
+
+  /**
+   * Creates a ValueObserver with the passed characteristics and returns a shared_ptr to that
+   * ValueObserver.
+   *
+   * @param name the name of the new ValueObserver.
+   * @param description a brief description of what the ValueObserver is used for.
+   * @param unit the unit of metric values following https://unitsofmeasure.org/ucum.html.
+   * @param enabled a boolean value that turns on or off the metric instrument.
+   * @param callback the function to be observed by the instrument.
+   * @return a shared pointer to the created ValueObserver.
+   * @throws NullPointerException if {@code name} is null
+   * @throws IllegalArgumentException if a different metric by the same name exists in this meter.
+   * @throws IllegalArgumentException if the {@code name} does not match spec requirements.
+   */
+  virtual opentelemetry::nostd::shared_ptr<AsynchronousInstrument<T>> NewValueObserver(
+      nostd::string_view name,
+      nostd::string_view description,
+      nostd::string_view unit,
+      const bool enabled,
+      void (*callback)(ObserverResult<T>)) = 0;
+
+  /**
+   * Utility method that allows users to atomically record measurements to a set of
+   * synchronous metric instruments with a common set of labels.
+   *
+   * @param labels the set of labels to associate with this recorder.
+   * @param values a span of pairs where the first element of the pair is a metric instrument
+   * to record to, and the second element is the value to update that instrument with.
+   */
+  virtual void RecordBatch(
+      const trace::KeyValueIterable &labels,
+      const nostd::span<std::pair<nostd::shared_ptr<SynchronousInstrument<T>>,
+          T>> values) noexcept
+  {
+    // No-op
+  }
+
+  /**
+   * Utility method that allows users to atomically record measurements to a set of
+   * synchronous metric instruments with a common set of labels.
+   *
+   * This overloads the BatchRecord function to allow {@code labels} to be of type initializer list.
+   *
+   * @param labels an initializer list holding labels to associate with this recording.
+   * @param values a span of pairs where the first element of the pair is a metric instrument
+   * to record to, and the second element is the value to update that instrument with.
+   */
+  void RecordBatch(
+      std::initializer_list<std::pair<nostd::string_view, nostd::string_view>> labels,
+      const nostd::span<std::pair<nostd::shared_ptr<SynchronousInstrument<T>>,
+          T>> values) noexcept
+  {
+    // No-op
```

Comment:
All non-virtual versions of `RecordBatch` should transform arguments and redirect to the virtual version of `RecordBatch`.

Reply:
Ok I see. Working on that now! Thanks!